streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -1048,4 +1042,28 @@ public String toString(final String indent) {
     Set<TaskId> lockedTaskDirectories() {
         return Collections.unmodifiableSet(lockedTaskDirectories);
     }
+
+    public static void executeAndMaybeSwallow(final boolean clean,
+                                              final Runnable runnable,
+                                              final java.util.function.Consumer<RuntimeException> actionIfClean,
+                                              final java.util.function.Consumer<RuntimeException> actionIfNotClean) {
```

Comment:
At first thought, the not `clean` case normally just do some logs. with only the log string different,  I would like to still use the `executeAndMaybeSwallow ` in `Task.scala`, but since callers may need different log levels or they may want do more than just a log, so just pass a more generic argument `actionIfNotClean`.

Reply:
I am wondering if adding this method to `TaskManager` is the best choice (cf https://issues.apache.org/jira/browse/KAFKA-10055). Thoughts?

Reply:
@abbccdda :(
@mjsax 
I have similar thoughts when doing this, but I don't know enough context to decide whether this method is general enough to be extracted out to a utility class. If there are more similar static methods in other places, then I think we should put it in the utility class.

Reply:
I am also fine to just merge this as-is, and refactor via KAFKA-10055.

\cc @abbccdda WDYT?

Reply:
I'm fine as well, will make a reference to 10055 of this PR